src/wp-includes/functions.php

```diff
@@ -6239,15 +6239,16 @@ function validate_file( $file, $allowed_files = array() ) {
  *
  * @since 2.6.0
  *
- * @param string|bool $force Optional. Whether to force SSL in admin screens. Default null.
+ * @param string|bool|null $force Optional. Whether to force SSL in admin screens. Default null.
  * @return bool True if forced, false if not forced.
  */
 function force_ssl_admin( $force = null ) {
 	static $forced = false;
 
 	if ( ! is_null( $force ) ) {
+		// Check if $force is a boolean, typecast to boolean if it's not.
 		$old_forced = $forced;
-		$forced     = $force;
+		$forced     = (bool) $force; // Always cast to boolean.
```

Comment:
The additional comments here are superfluous, they don't really add any value as the casting is self-explanatory.

Reply:
Hi @swissspidy, I was thinking of updating the logic a bit.

What do you think about this?

```
$forced = wp_validate_boolean( $force );
```

This will give this function a far better flexibility in terms of string inputs as well.

Please let me know your thoughts on this. Thanks.

Reply:
That sounds overkill to me.

Let‘s keep the cast as is, but remove these comments

Reply:
Sure, pushing the code. Thanks for the feedback.